Common/src/linear_algebra/CSysVector.cpp

```diff
@@ -52,7 +52,7 @@ CSysVector<ScalarType>::CSysVector(const unsigned long & size, const ScalarType
   nVar = 1;
 
   /*--- Check for invalid size, then allocate memory and initialize values ---*/
-  if ( (nElm <= 0) || (nElm >= UINT_MAX) ) {
+  if ( (nElm < 0) || (nElm >= UINT_MAX) ) {
```

Comment:
nElm is not going to be less than 0, please change UINT_MAX to ULONG_MAX to be consistent with the other constructors